cmd/snap-seccomp/main.go

```diff
@@ -182,6 +183,32 @@ package main
 // #ifndef PTRACE_SETFPXREGS
 // #define PTRACE_SETFPXREGS 19
 // #endif
+//
+// /* Define missing kcmp constants */
+// #ifndef KCMP_FILE
+// #define KCMP_FILE 0
+// #endif
+// #ifndef KCMP_VM
+// #define KCMP_VM 1
+// #endif
+// #ifndef KCMP_FILES
+// #define KCMP_FILES 2
+// #endif
+// #ifndef KCMP_FS
+// #define KCMP_FS 3
+// #endif
+// #ifndef KCMP_SIGHAND
+// #define KCMP_SIGHAND 4
+// #endif
+// #ifndef KCMP_IO
+// #define KCMP_IO 5
+// #endif
+// #ifndef KCMP_SYSVSEM
+// #define KCMP_SYSVSEM 6
+// #endif
+// #ifndef KCMP_EPOLL_TFD
+// #define KCMP_EPOLL_TFD 7
+// #endif
```

Comment:
In the upstream kernel headers, these symbols come from the `kcmp_type` enum:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/kcmp.h

They are not preprocessor macros, so all of these `#ifndef` blocks will be included and we'll shadow the enumeration values with `#define`s to integers. So we're always ignoring the enumeration values from the kernel with this.

Maybe a better option would be to use the macros in `<linux/version.h>` to decide whether we've got a new enough kernel to rely on `<linux/kcmp.h>`, and then declare the enumeration (or macros) if not.

Reply:
Ok, so `<linux/kcm.h>` was added in 3.19

Reply:
And `KCMP_EPOLL_TFD` was added in 4.13

Reply:
Given we only care about the `KCMP_FILE` constant for now, we can probably ignore `KCMP_EPOLL_TFD`.

So the fallback could look something like:

```
#include <linux/version.h>
#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 19, 0)
#  include <linux/kcmp.h>
#else
#  define KCMP_FILE 0
#  define KCMP_VM 1
...
#endif
```

The man page says kcmp() was added in 3.5, so I'm not sure whether the version check is fully correct. It's far enough back that I'm not sure it really matters though.

Reply:
Right, 3.19 made a move to uapi